erdos/src/dataflow/operators/filter.rs

```diff
@@ -64,3 +69,31 @@ where
 
     fn on_watermark(&mut self, _ctx: &mut OneInOneOutContext<(), D1>) {}
 }
+
+// Extension trait for FilterOperator
+pub trait Filter<D1>
+where
+    D1: Data + for<'a> Deserialize<'a>,
+{
+    fn filter<F: 'static + Fn(&D1) -> bool + Send + Sync + Clone>(
+        &self,
+        filter_fn: F,
+    ) -> Stream<D1>;
+}
+
+impl<D1> Filter<D1> for Stream<D1>
+where
+    D1: Data + for<'a> Deserialize<'a>,
+{
+    fn filter<F: 'static + Fn(&D1) -> bool + Send + Sync + Clone>(
+        &self,
+        filter_fn: F,
+    ) -> Stream<D1> {
+        crate::connect_one_in_one_out(
+            move || -> FilterOperator<D1> { FilterOperator::new(filter_fn.clone()) },
+            || {},
+            OperatorConfig::new().name("FilterOperator"),
```

Comment:
Can you please add the name of the stream from `self` to the name of the operator? It might help us in debugging this at a future point.